util/report_utils/compare_results.rb

```diff
@@ -4,55 +4,69 @@
 # TODO: accept release names as optional arguments for A / B
 
 require "csv"
+require "../../tests/helpers/perf_results_helper.rb"
+include PerfResultsHelper # rubocop:disable Style/MixinUsage
 
-def diff(a, b) # rubocop:disable Naming/UncommunicativeMethodParamName
-  result = ((b.to_f - a.to_f) / a.to_f) * 100
+raise Exception, "you must provide two csv files to compare" unless ARGV[1] && ARGV[2]
+
+result_a_path = ARGV[0]
+result_b_path = ARGV[1]
+
+def diff(result_a, result_b)
+  result = ((result_b.to_f - result_a.to_f) / result_a.to_f) * 100
   "#{result.round(2)}%"
 end
 
-raise Exception, "you must provide two csv files to compare" unless ARGV[1]
+# TODO: refactor
+# rubocop: disable Metrics/AbcSize
+def write_csv(output_path, result_a, result_b)
+  # TODO: use release names
+  CSV.open(output_path.to_s, "wb") do |csv|
+    csv << ["", "$RELEASE_A_NAME ($RELEASE_A_NUMBER)", "", "",
+            "$RELEASE_B_NAME ($RELEASE_B_NUMBER)", "", ""]
+    csv << ["Duration", "max ms", "mean ms", "std dev", "max ms", "mean ms",
+            "std dev", "% (mean) diff"]
 
-result_a_path = ARGV[0]
-result_a_name = File.basename(result_a_path, ".*")
+    csv << ["Total", result_a[1][1], result_a[1][2], result_a[1][3],
```

Comment:
This seams a little repetitive. Can we wrap this in an block?
Maybe something like:
```
headings = ["Total", "catalog", "filemeta plugins", ...]
headings.each_with_index do |item, index|
  csv << [item, result_a[index+1][1], result_a[index+1][2], result_a[index+1][3],
               result_b[index+1][1], result_b[index+1][2], result_b[index+1][3],
               diff(result_a[index+1][2], result_b[index+1][2])]
end

Reply:
Nice! Updated...